test/arrayops.jl

```diff
@@ -1153,17 +1153,17 @@ end
     # issue #5177
 
     c = fill(1,2,3,4)
-    m1 = mapslices(x-> fill(1,2,3), c, dims=[1,2])
-    m2 = mapslices(x-> fill(1,2,4), c, dims=[1,3])
-    m3 = mapslices(x-> fill(1,3,4), c, dims=[2,3])
+    m1 = mapslices(Returns(fill(1,2,3)), c, dims=[1,2])
+    m2 = mapslices(Returns(fill(1,2,4)), c, dims=[1,3])
+    m3 = mapslices(Returns(fill(1,3,4)), c, dims=[2,3])
     @test size(m1) == size(m2) == size(m3) == size(c)
 
-    n1 = mapslices(x-> fill(1,6), c, dims=[1,2])
-    n2 = mapslices(x-> fill(1,6), c, dims=[1,3])
-    n3 = mapslices(x-> fill(1,6), c, dims=[2,3])
-    n1a = mapslices(x-> fill(1,1,6), c, dims=[1,2])
-    n2a = mapslices(x-> fill(1,1,6), c, dims=[1,3])
-    n3a = mapslices(x-> fill(1,1,6), c, dims=[2,3])
+    n1 =  mapslices(Returns(fill(1,6)  ), c, dims=[1,2])
+    n2 =  mapslices(Returns(fill(1,6)  ), c, dims=[1,3])
+    n3 =  mapslices(Returns(fill(1,6)  ), c, dims=[2,3])
+    n1a = mapslices(Returns(fill(1,1,6)), c, dims=[1,2])
+    n2a = mapslices(Returns(fill(1,1,6)), c, dims=[1,3])
+    n3a = mapslices(Returns(fill(1,1,6)), c, dims=[2,3])
```

Comment:
Since these are not functionally equivalent, I would prefer leaving this as-is.

Reply:
So you say that whether only one `fill(...)` or a couple of `fill(...)` are allocated could be a detail that is relevant here?

Reply:
It's probably not a big deal here, but it's often easy to miss the purpose of a test like this if you did not write it.

Reply:
Done!